// Copyright 2018 Google LLC. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Representation of IPA symbol table as a protocol message.

syntax = "proto2";

package fonbund;

// Unicode representation of IPA symbol expressed in Universal
// Character Set (UCS).
message IpaSymbol {
  // Decimal code describing the symbol (e.g. 97 for "a").
  optional int32 decimal_code = 1;

  // Hexadecimal code for the symbol represented as string (e.g. "0061"
  // for "a").
  optional string hex_code = 2;

  // Example rendering(s), as string (e.g. for "breathy voiced" diacritic
  // this will be "b̤,a̤").
  optional string example_rendering = 3;

  // Description for this symbol (e.g. "open front unrounded vowel").
  optional string description = 4;

  // Some useful precomposed characters don't fit into the "official" IPA.
  // For such characters the following flag should be enabled.
  optional bool unofficial = 5;

  // The sequence of decimal codes this symbol maps to.
  repeated int32 maps_to_decimal_code = 6;

  // The sequence of hex codes this symbol maps to.
  repeated string maps_to_hex_code = 7;
}

message IpaSymbols {
  // List of IPA symbols.
  repeated IpaSymbol symbol = 1;

  // Allow precomposed characters, such as 'ʧ', that don't fit into the
  // official IPA.
  optional bool allow_unofficial_characters = 2;
}
